src/components/appSwitch/AppSwitch.js

```diff
@@ -66,6 +67,7 @@ const AppSwitch = (props: LoadingProps) => {
   const store = SimpleStore.useStore()
   const [showErrorDialog] = useErrorDialog()
   const [ready, setReady] = useState(false)
+  const { updateIsLoggedIn, updateIsLoggedInCitizen } = useContext(UserContext)
```

Comment:
no need to use setters per each vars, just:

```suggestion
  const { setUserConext } = useContext(UserContext)
```

Reply:
setUserConext should merge current context fields with the new ones

Also you could create resetUserContext(...vars: string[]) which resets vars values to the default ones (optionally, you could just do setUserContext(isLoggedIn: false)